// storybook-addon-html-document/lib/decorators/react.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.withHTMLDOC = void 0;

var _react = _interopRequireWildcard(require("react"));

var _global = require("global");

var _addons = require("@storybook/addons");

var _shared = require("../shared");

var _ = require(".");

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var div = document.createElement('div');
div.style.position = 'absolute';
div.style.top = '0';
div.style.left = '0';
div.style.right = '0';
div.style.bottom = '0';
var iframe = document.createElement('iframe');
iframe.setAttribute('frameborder', 0);
iframe.setAttribute('marginheight', 0);
iframe.setAttribute('marginheight', 0);
iframe.setAttribute('allowfullscreen', true);
iframe.setAttribute('scrolling', 'yes');
iframe.style.width = '100%';
iframe.style.height = '100%';
div.appendChild(iframe);

var ReactWrapper = function ReactWrapper(_ref) {
  var setCode = _ref.setCode,
      children = _ref.children;

  var wrapperRef = _react["default"].createRef();

  (0, _react.useEffect)(function () {
    setCode(wrapperRef.current.innerHTML);
  });
  return /*#__PURE__*/_react["default"].createElement("div", {
    ref: wrapperRef
  }, children);
};

var HtmlWrapper = function HtmlWrapper(_ref2) {
  var _React$createElement;

  var srcDoc = _ref2.srcDoc;
  return /*#__PURE__*/_react["default"].createElement("div", {
    style: {
      position: 'absolute',
      top: 0,
      left: 0,
      right: 0,
      bottom: 0
    }
  }, /*#__PURE__*/_react["default"].createElement("iframe", (_React$createElement = {
    style: {
      width: '100%',
      height: '100%'
    },
    frameBorder: 0,
    marginHeight: 0
  }, _defineProperty(_React$createElement, "marginHeight", 0), _defineProperty(_React$createElement, "allowFullScreen", true), _defineProperty(_React$createElement, "scrolling", "yes"), _defineProperty(_React$createElement, "srcDoc", srcDoc), _React$createElement)));
};

var withHTMLDOC = (0, _addons.makeDecorator)(_objectSpread({}, _.parameters, {
  wrapper: function wrapper(getStory, context, _ref3) {
    var _ref3$options = _ref3.options,
        options = _ref3$options === void 0 ? {} : _ref3$options;

    var channel = _addons.addons.getChannel();

    var element = getStory();

    if (_react["default"].isValidElement(element)) {
      return /*#__PURE__*/_react["default"].createElement(ReactWrapper, {
        setCode: function setCode(html) {
          channel.emit(_shared.EVENT_CODE_RECEIVED, {
            html: html,
            options: options
          });
        }
      }, element);
    } else {
      var html;

      if (typeof element === 'string') {
        html = element;
      } else if (element instanceof _global.Node) {
        html = element.outerHTML;
      }

      channel.emit(_shared.EVENT_CODE_RECEIVED, {
        html: html,
        options: options
      });
      return /*#__PURE__*/_react["default"].createElement(HtmlWrapper, {
        srcDoc: html
      });
    }
  }
}));
exports.withHTMLDOC = withHTMLDOC;

if (module && module.hot && module.hot.decline) {
  module.hot.decline();
}
